src/v/bytes/iobuf.h

```diff
@@ -389,14 +403,31 @@ iobuf::append(const uint8_t* src, size_t len) {
     }
     append(std::make_unique<fragment>(std::move(b)));
 }
+
 /// appends the contents of buffer; might pack values into existing space
-inline void iobuf::append(iobuf o) {
-    while (!o._frags.empty()) {
-        fragment* f = &o._frags.front();
-        o._frags.pop_front();
-        append(f->share());
-        details::dispose_io_fragment(f);
+inline void iobuf::append(iobuf&& o) {
+    for (auto& f : o._frags) {
+        auto fsize = f.size();
+
+        if (unlikely(fsize)) {
```

Comment:
Should this be `fsize == 0`?

Reply:
Yeah, sorry did a refactor before pushing.